### PR TITLE
ADMIN-332 fix cannot read properties of undefined thrown by sentry

### PR DESCRIPTION
**Pull request by ahadziii**
## Description

went through the issues section in sentry and fixed the "cannot read property of undefined errors"

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?
yes this has been tested on my local machine

### CODE REVIEW

File: web-react-ts/src/auth/SetPermissions.tsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ const SetPermissions = ({
         fellowship: data.memberByEmail?.fellowship.id,
         bacenta: data.memberByEmail?.fellowship?.bacenta?.id,
         council:
-          data.memberByEmail?.fellowship?.bacenta.constituency?.council.id,
+          data.memberByEmail?.fellowship?.bacenta.constituency?.council?.id,
```
Comment (jaedag) [inline web-react-ts/src/auth/SetPermissions.tsx:47]: I don't understand this bit. How would a member not have a council

File: web-react-ts/src/components/formik/ImageUpload.tsx [hunk 2]
```diff
@@ -37,7 +37,7 @@ const ImageUpload = (props: ImageUploadProps) => {
     data.append('upload_preset', uploadPreset || '')
     data.append(
       'public_id',
-      `${username}-${currentUser.id}/${date}_${files[0].name}`
+      `${username}-${currentUser.id}/${date}_${files[0]?.name}`
```
Comment (jaedag) [inline web-react-ts/src/components/formik/ImageUpload.tsx:40]: Is there a reason why the file would not have a name?

File: web-react-ts/src/pages/arrivals/BacentaArrivals.tsx [hunk 3]
```diff
@@ -73,7 +73,7 @@ const BacentaArrivals = () => {
   useEffect(() => handleClose(), [])
 
   const END_TIME_IN_MS = new Date(
-    getTodayTime(bacenta?.stream.arrivalEndTime)
+    getTodayTime(bacenta?.stream?.arrivalEndTime)
```
Comment (jaedag) [inline web-react-ts/src/pages/arrivals/BacentaArrivals.tsx:76]: Is there a reason why the arrivalEndTime would not be there for a stream? And if so which stream is this? I would like to know

File: web-react-ts/src/pages/directory/display/DetailsBacenta.tsx [hunk 4]
```diff
@@ -17,7 +17,7 @@ const DetailsBacenta = () => {
   const bacenta = data?.bacentas[0]
 
   let breadcrumb = [
-    bacenta?.constituency.council,
+    bacenta?.constituency?.council,
```
Comment (jaedag) [inline web-react-ts/src/pages/directory/display/DetailsBacenta.tsx:20]: How would there be a constituency without a council? 
Are you able to understand how that would happen?

File: web-react-ts/src/pages/services/graphs/TrendsMenu.tsx [hunk 5]
```diff
@@ -28,7 +28,7 @@ const TrendsMenu = () => {
             iconComponent={Charts}
             title="Graphs"
             color="members"
-            onClick={() => navigate(`/${churchType.toLowerCase()}/graphs`)}
+            onClick={() => navigate(`/${churchType?.toLowerCase()}/graphs`)}
```
Comment (jaedag) [inline web-react-ts/src/pages/services/graphs/TrendsMenu.tsx:31]: If there is no churchType, the link will lead to a 404. Are you able to understand why there would be no churchType?

File: web-react-ts/src/pages/services/graphs/TrendsMenu.tsx [hunk 6]
```diff
@@ -37,7 +37,7 @@ const TrendsMenu = () => {
             color="quick-facts"
             onClick={() =>
               navigate(
-                `/quick-facts/this-month/${church.__typename.toLowerCase()}`
+                `/quick-facts/this-month/${church?.__typename.toLowerCase()}`
```
Comment (jaedag) [inline web-react-ts/src/pages/services/graphs/TrendsMenu.tsx:40]: If there is no church.typename, the navlink will lead to a 404. Are you able to understand why there is no church typename?

**Review by jaedag — CHANGES_REQUESTED** [review]

File: web-react-ts/src/auth/SetPermissions.tsx (hunk 1 above)
Comment (ahadziii) [reply web-react-ts/src/auth/SetPermissions.tsx:47]: I don't know if it is graphql thing. but I realised while working sometime ago that some of the data is undefined and after some seconds the data loads with values. I am sure thats what is happening

File: web-react-ts/src/pages/directory/display/DetailsBacenta.tsx (hunk 4 above)
Comment (ahadziii) [reply web-react-ts/src/pages/directory/display/DetailsBacenta.tsx:20]: I think the data in use query is undefined  on the first render cycle

**Pull request closed without merging**